source/mainwindow.cpp

```diff
@@ -236,7 +241,30 @@ void MainWindow::saveAs(QString x) {
   if (currentFile != "" && !image.isNull())
     image.save(currentFile + "." + x, x.toUpper().toLocal8Bit().constData());
 }
-
+void MainWindow::rename() {
+  if(currentFile != "" && !image.isNull()) {
+      bool ok;
+      QFileInfo file(currentFile);
+      QString currentName = file.fileName();
+      QString text = QInputDialog::getText(this, tr("Rename"),
+                                           tr("Name for the file:"), QLineEdit::Normal,
+                                            currentName, &ok);
+      if(ok && !text.isEmpty()) {
```

Comment:
Currently there's an if else inside another if condition. This generally increases cognitive complexity.

So we can rearrange the conditions a bit to make it such that these conditions are not nested.
```
if(!ok || text.isEmpty()) return;
```
Also, try to avoid braces `{}` if there's just one single command inside an `if` block.

Reply:
This one last change, and we are good to merge.